server.go

```diff
@@ -1133,6 +1133,29 @@ out:
 
 			srvrLog.Infof("Detected new external IP address %s", ip)
 
+			// After detecting the new external IP try to re-create the
+			// port forwardings for the used ports.
+			for _, port := range forwardedPorts {
+				// first remove a possibly still existing forward
+				if err := s.natTraversal.DeletePortMapping(port); err != nil {
+					srvrLog.Debugf("Unable to remove forward %d: %v", port, err)
+					// assume the port is no longer forwarded
+					s.natTraversal.DeleteForwardedPort(port)
```

Comment:
this seems like it should be done in the success case, no? how do we know the port mapping was actually removed if an error was returned?